receiver/httpcheckreceiver/scraper.go

```diff
@@ -85,6 +87,22 @@ func (h *httpcheckScraper) scrape(ctx context.Context) (pmetric.Metrics, error)
 					h.mb.RecordHttpcheckStatusDataPoint(now, int64(0), h.cfg.Targets[targetIndex].Endpoint, int64(statusCode), req.Method, class)
 				}
 			}
+
+			substring := h.cfg.Targets[targetIndex].Substring
+			if err == nil && len(substring) > 0 {
+				if body, err := io.ReadAll(resp.Body); err != nil {
+					h.settings.Logger.Error("failed to read response", zap.Error(err))
+				} else {
+					defer resp.Body.Close()
+
+					if strings.Contains(string(body), substring) {
```

Comment:
I note here that `string(body)` does not care for encoding. Not sure if this is a problem.

Reply:
Very detailed observation! At that time, i was only considering plain text responses for HTTP, without considering encoding formats like base64 or others. If we need to consider encoding formats, we may have to take into account multiple encoding methods?

Reply:
yeah I am all for you to make this incremental improvement, and we can always go back and learn more about what to expect. I am more worried of double byte encoded strings such as UTF-16 encoded strings.

Reply:
we could maybe check the encoding on the header?

Reply:
and based on the header add the decode method? like a switch

Reply:
here we go, that's how it should be done. That's probably something to keep in mind for a subsequent PR.